src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs

```diff
@@ -40,6 +40,8 @@ public abstract class RazorPage : IRazorPage
         private TagHelperAttributeInfo _tagHelperAttributeInfo;
         private HtmlContentWrapperTextWriter _valueBuffer;
         private IViewBufferScope _bufferScope;
+        private bool _ignoreBody;
+        private readonly HashSet<string> _ignoredSections = new HashSet<string>();
```

Comment:
Could we lazily initialize this? We could avoid allocating the `HashSet` if this feature isn't used. 
```C#
public void IgnoreSection(string name)
{
   ...
   if (_ignoredSections == null)
   {
      _ignoredSections = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
    }
    ...
}
```